lib/fabrik/DirectorInstance.js

```diff
@@ -249,12 +288,21 @@ class DirectorInstance extends BaseInstance {
       .then(() => this.manager.deleteBinding(this.deploymentName, params.binding_id));
   }
 
-  buildSecurityGroupRules() {
+  buildIpRules() {
+
+    function getPorts(first, last) {
+      let ports = [];
+      for (let i = first; i <= last; i++) {
+        ports.push(i);
+      }
+      return ports;
+    }
+
     return _.map(this.manager.getNetwork(this.networkSegmentIndex), net => {
       return {
         protocol: 'tcp',
-        destination: `${_.first(net.static)}-${_.last(net.static)}`,
-        ports: '1024-65535'
+        ips: net.static,
+        ports: getPorts(1024, 65535)
```

Comment:
instead you could have written `ports: [1024, 65535]`. You are anyways considering first and last. So why populate such a huge array. If your problem is that semantically its not right, then you set attrib `port` - for a single port in docker & `port_range` with value `1024-65535` so that way you pick based on port or port_range and you can avoid this huge ports array.

Reply:
will set [1024, 65535]